accounting/accounting_cli_functions.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+###############################################################
+# Copyright 2020 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+import sqlite3
+import argparse
+import time
+import sys
+
+import pandas as pd
+
+
+def view_user(conn, user):
+    try:
+        # get the information pertaining to a user in the Accounting DB
+        select_stmt = "SELECT * FROM association_table where user_name=?"
+        dataframe = pd.read_sql_query(select_stmt, conn, params=(user,))
+        # if the length of dataframe is 0, that means
+        # the user specified was not found in the table
+        if len(dataframe.index) == 0:
+            print("User not found in association_table")
+        else:
+            print(dataframe)
+    except pd.io.sql.DatabaseError as e_database_error:
+        print(e_database_error)
+
+
+def add_user(conn, user_acct_info):
+    # make sure all of the fields are entered
+    if len(user_acct_info) != 7:
+        print("Not all fields were entered")
+        sys.exit(1)
+
+    # insert the user values into the database
+    try:
+        conn.execute(
+            """
+            INSERT INTO association_table (
+                creation_time,
+                mod_time,
+                deleted,
+                user_name,
+                admin_level,
+                account,
+                parent_acct,
+                shares,
+                max_jobs,
+                max_wall_pj
+            )
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+            """,
+            (
+                int(time.time()),
+                int(time.time()),
+                0,
+                user_acct_info[0],
+                user_acct_info[1],
+                user_acct_info[2],
+                user_acct_info[3],
+                user_acct_info[4],
+                user_acct_info[5],
+                user_acct_info[6],
```

Comment:
The ordering of the elements within the `add_user` list depends on the order that the user provides them in, but the `add_user` makes assumptions about their ordering.  If the user provides the args in the same order as the help message, all is good, but if you swap any arguments, the database will have the wrong values:

```
# sherbein at Stephens-MacBook-Pro.local in ~/Repositories/flux-framework/flux-accounting on git:front-end-tools ✖︎ [19:42:39]
→ PYTHONPATH=. ./accounting/accounting_cli.py add-user --user_name stephen --admin_level 1 --account foo --parent_acct NULL --shares 5 --max_jobs 6 --max_wall_pj 7

# sherbein at Stephens-MacBook-Pro.local in ~/Repositories/flux-framework/flux-accounting on git:front-end-tools ✖︎ [19:43:01]
→ PYTHONPATH=. ./accounting/accounting_cli.py view-user stephen
   id_assoc  creation_time    mod_time  deleted user_name  admin_level account parent_acct  shares  max_jobs  max_wall_pj
0         1     1590201763  1590201763        0   stephen            1     foo        NULL       5         6            7

# sherbein at Stephens-MacBook-Pro.local in ~/Repositories/flux-framework/flux-accounting on git:front-end-tools ✖︎ [19:43:34]
→ PYTHONPATH=. ./accounting/accounting_cli.py add-user --user_name mark --account bar --admin_level 2 --parent_acct NULL --shares 5 --max_jobs 6 --max_wall_pj 7

# sherbein at Stephens-MacBook-Pro.local in ~/Repositories/flux-framework/flux-accounting on git:front-end-tools ✖︎ [19:43:43]
→ PYTHONPATH=. ./accounting/accounting_cli.py view-user mark
   id_assoc  creation_time    mod_time  deleted user_name admin_level account parent_acct  shares  max_jobs  max_wall_pj
0         2     1590201823  1590201823        0      mark         bar       2        NULL       5         6            7
```

I recommend either making all of these arguments positional arguments to the function, or use `**kwargs`.  You will also need to not have every `dest=add_user` in the argparser.

Reply:
Additional idea, check inputs for validity?  Validity may not be perfect, but simple things like is a number, is a string with alphanumeric, etc.